### PR TITLE
[port] Add a facility to parse and validate script bitfields

### PR DESCRIPTION
**Pull request by sickpig**
Bitfield are little endian and has a size proportional to
the number of bits they represent. The decoder validates that
there are no out of bound bits.

This is a port of reviews.bitcoinabc.org/D3735

~~Built on top of #1901~~

This is a prereq for porting Schnorr multi sig from ABC.

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -447,6 +447,8 @@ libbitcoin_consensus_a_SOURCES = \
   pubkey.cpp \
   pubkey.h \
   script/bitcoinconsensus.cpp \
+  script/bitfield.cpp \
+  script/bitfield.h \
   script/interpreter.cpp \
```
Comment (ptschip) [inline src/Makefile.am:453]: add these files to .formatted-files   also add bitfeild_tests.cpp

Comment (sickpig) [reply src/Makefile.am:453]: Good point. Will do

Comment (sickpig) [reply src/Makefile.am:453]: done

File: src/script/script_error.h [hunk 2]
```diff
@@ -80,4 +80,21 @@ typedef enum ScriptError_t {
 
 const char *ScriptErrorString(const ScriptError error);
 
+namespace
+{
```
Comment (dagurval) [inline src/script/script_error.h]: ^^ remove anon namespace from header file

Comment (sickpig) [reply src/script/script_error.h]: will do

**Review by dagurval — APPROVED** [review]

**Comment by sickpig** [comment]
rebased on top of `dev`

**Comment by ptschip** [comment]
Looks like one of the unit tests is failing

**Comment by sickpig** [comment]
> Looks like one of the unit tests is failing

failure seems to be a travis hiccup rather than anything else, restarted the travis job and it passed. I tried locally mulitple times and I wasn't able to reproduce the problem.

**Pull request merged**